Sources/Commands/Options.swift

```diff
@@ -246,14 +246,29 @@ public struct SwiftToolOptions: ParsableArguments {
     @Flag(name: [.long, .customLong("disable-automatic-resolution")], help: "Disable automatic resolution if Package.resolved file is out-of-date")
     var forceResolvedVersions: Bool = false
 
-    @Flag(name: .customLong("index-store"), inversion: .prefixedEnableDisable, help: "Enable or disable  indexing-while-building feature")
-    var indexStoreEnable: Bool?
+    // @Flag works best when there is a default value present
+    // if true, false aren't enough and a third state is needed
+    // nil should not be the goto. Instead create an enum
+    enum StoreMode: EnumerableFlag {
+        case autoIndexStore
+        case enableIndexStore
+        case disableIndexStore
         
-    /// The mode to use for indexing-while-building feature.
-    var indexStore: BuildParameters.IndexStoreMode {
-        guard let enable = indexStoreEnable else { return .auto }
-        return enable ? .on : .off
+        /// The mode to use for indexing-while-building feature.
+        var indexStoreMode: BuildParameters.IndexStoreMode {
+            switch self {
+            case .autoIndexStore:
+                return .auto
+            case .enableIndexStore:
+                return .on
+            case .disableIndexStore:
+                return .off
+            }
+        }
     }
+    
+    @Flag(help: "Enable or disable indexing-while-building feature")
+    var indexStoreMode: StoreMode = .autoIndexStore
```

Comment:
previously this was ` .customLong("index-store")`, did we loose that in the process?

Reply:
related, can you compare the previous options with the current ones, I suspect we are potentially breaking users with this change

Reply:
We also had `inversion: .prefixedEnableDisable` giving us `enable-index-store`, `disable-index-store`. But when using `EnumerableFlag` the name for the flag comes from the cases in the enum and not the name of the var so there's no need for the custom name and using the inversion

Reply:
okay, so to confirm are the flags before this change and after this change `enable-index-store` and `disable-index-store`?